src/libraries/System.Private.CoreLib/src/System/Index.cs

```diff
@@ -138,6 +143,15 @@ public override string ToString()
             return ((uint)Value).ToString();
         }
 
+        private static void ThrowValueArgumentOutOfRange_NeedNonNegNumException()
+        {
+#if SYSTEM_PRIVATE_CORELIB
+            throw new ArgumentOutOfRangeException("value", SR.ArgumentOutOfRange_NeedNonNegNum);
+#else
+            throw new ArgumentOutOfRangeException("value", "value must be non-negative");
```

Comment:
Don't our resources in source generators get localized?  Is the thinking here it doesn't matter because such an exception will never occur in our source generators?

Reply:
Generally exceptions in source generators are due to bugs (so not meant to be customer-facing), and we have no guarantee that there's a resource with the right name in the assembly that includes this file. As a result, I felt that inlining the message would provide a good enough experience for the situation.